test/unittest/models/unittest_models_recurrent.cpp

```diff
@@ -175,46 +175,46 @@ static std::unique_ptr<NeuralNetwork> makeStackedLSTM() {
   return nn;
 }
 
-// static std::unique_ptr<NeuralNetwork> makeSingleBidirectionalLSTM() {
-//   std::unique_ptr<NeuralNetwork> nn(new NeuralNetwork());
-//   nn->setProperty({"batch_size=3"});
-
-//   auto outer_graph = makeGraph({
-//     {"input", {"name=input", "input_shape=1:2:2"}},
-//     {"lstm",
-//      {"name=a1", "unit=2", "integrate_bias=false", "return_sequences=true",
-//       "bidirectional=true"}},
-//     {"mse", {"name=loss", "input_layers=a1"}},
-//   });
-//   for (auto &node : outer_graph) {
-//     nn->addLayer(node);
-//   }
-
-//   nn->setOptimizer(ml::train::createOptimizer("sgd", {"learning_rate =
-//   0.1"})); return nn;
-// }
-
-// static std::unique_ptr<NeuralNetwork> makeStackedBidirectionalLSTM() {
-//   std::unique_ptr<NeuralNetwork> nn(new NeuralNetwork());
-//   nn->setProperty({"batch_size=3"});
-
-//   auto outer_graph = makeGraph({
-//     {"input", {"name=input", "input_shape=1:2:2"}},
-//     {"lstm",
-//      {"name=a1", "unit=2", "integrate_bias=false", "return_sequences=true",
-//       "bidirectional=true"}},
-//     {"lstm",
-//      {"name=a2", "unit=2", "integrate_bias=false", "return_sequences=true",
-//       "bidirectional=true"}},
-//     {"mse", {"name=loss"}},
-//   });
-//   for (auto &node : outer_graph) {
-//     nn->addLayer(node);
-//   }
-
-//   nn->setOptimizer(ml::train::createOptimizer("sgd", {"learning_rate =
-//   0.1"})); return nn;
-// }
+static std::unique_ptr<NeuralNetwork> makeSingleBidirectionalLSTM() {
+  std::unique_ptr<NeuralNetwork> nn(new NeuralNetwork());
+  nn->setProperty({"batch_size=3"});
+
+  auto outer_graph = makeGraph({
+    {"input", {"name=input", "input_shape=1:2:2"}},
+    {"lstm",
+     {"name=a1", "unit=2", "integrate_bias=false", "return_sequences=true",
+      "bidirectional=true"}},
+    {"mse", {"name=loss"}},
+  });
+  for (auto &node : outer_graph) {
+    nn->addLayer(node);
+  }
+
+  nn->setOptimizer(ml::train::createOptimizer("sgd", {"learning_rate = 0.1"}));
+  return nn;
+}
+
+static std::unique_ptr<NeuralNetwork> makeStackedBidirectionalLSTM() {
+  std::unique_ptr<NeuralNetwork> nn(new NeuralNetwork());
```

Comment:
How about using `make_unique()` ?
```suggestion
  auto nn = std::make_unique<NeuralNetwork>();
```

please check the reference: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3588.txt

> make_unique shares two advantages with make_shared (excluding the third advantage, increased
efficiency). First, unique_ptr<LongTypeName> up(new LongTypeName(args)) must
mention LongTypeName twice, while auto up = make_unique<LongTypeName>(args)
mentions it once. Second, make_unique prevents the unspecified-evaluation-order
leak triggered by expressions like foo(unique_ptr<X>(new X),
unique_ptr<Y>(new Y)). (Following the advice "never say new" is simpler than
"never say new, unless you immediately give it to a named unique_ptr".)

Reply:
Thank you for your notice :)